### PR TITLE
clickhouse-local enhancements

### PR DESCRIPTION
**Pull request by ludv1x**
Added --stacktrace option to clickhouse-local. [#METR-23765]
Added support of path config variable and ability to load clickhouse-server data. [#METR-20000]
Default database of clickhouse-local is changed to "_local" to avoid collisions with clickhouse-server.

Small fixes and enhancements of command line options (setting and limits). [#METR-23101]

### CODE REVIEW

File: dbms/src/Server/LocalServer.cpp [hunk 1]
```diff
@@ -215,19 +221,41 @@ void LocalServer::handleHelp(const std::string & name, const std::string & value
 }
 
 
+/// If path specified and isn't empty, try setup server enviroment and load existsing metadata
```
Comment (alexey-milovidov) [inline dbms/src/Server/LocalServer.cpp]: Typo.

File: dbms/src/Server/Server.cpp [hunk 2]
```diff
@@ -178,27 +178,33 @@ class TCPConnectionFactory : public Poco::Net::TCPServerConnectionFactory
...
 		throw Exception("path configuration parameter is empty");
 	if (path.back() != '/')
 		path += '/';
+	return path;
+}
+
+
+int Server::main(const std::vector<std::string> & args)
+{
+	Logger * log = &logger();
```
Comment (alexey-milovidov) [inline dbms/src/Server/Server.cpp]: Why this is necessary?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: dbms/src/Server/Server.cpp (hunk 2 above)
Comment (ludv1x) [inline dbms/src/Server/Server.cpp]: It's a matter of taste-)
It was strange to me that main server function begins with explicit low-level string manipulations.
Reverted.

Comment (alexey-milovidov) [inline dbms/src/Server/Server.cpp]: I mention only `Logger * log = &logger();` line.

Comment (ludv1x) [inline dbms/src/Server/Server.cpp]: The same line is in [master](https://github.com/yandex/ClickHouse/blob/master/dbms/src/Server/Server.cpp#L183) now. It is used in `Server::main` for [logging](https://github.com/yandex/ClickHouse/blob/master/dbms/src/Server/Server.cpp#L235).

Comment (alexey-milovidov) [inline dbms/src/Server/Server.cpp]: Now it is unused variable?

Comment (ludv1x) [inline dbms/src/Server/Server.cpp]: It is used now.

File: dbms/src/Server/LocalServer.cpp [hunk 3]
```diff
@@ -151,17 +161,13 @@ void LocalServer::defineOptions(Poco::Util::OptionSet& _options)
 		.callback(Poco::Util::OptionCallback<LocalServer>(this, &LocalServer::handleHelp)));
 
 #define DECLARE_SETTING(TYPE, NAME, DEFAULT) \
-	{ \
```
Comment (alexey-milovidov) [inline dbms/src/Server/LocalServer.cpp]: :-1:

Comment (alexey-milovidov) [inline dbms/src/Server/LocalServer.cpp]: https://github.com/yandex/ClickHouse/commit/bade1519ba543c355d2e4f607a474b418346043f
https://github.com/yandex/ClickHouse/commit/b6fb08c77fda64936bf7e7b1dc0d66d9ffef3eae

Comment (ludv1x) [inline dbms/src/Server/LocalServer.cpp]: Do you check that `--help` works properly after this chnage?

**Pull request merged**